File: springy-java/src/net/jangxyz/springy/Renderer.java
package net.jangxyz.springy;


public class Renderer {
	public interface Callback {
		void clear();		
		void drawEdge(Graph.Edge edge, Vector point1, Vector point2);
		void drawNode(Graph.Node node, Vector point);
		void onRenderStart();
		void onRenderStop();
	}

	ForceDirectedLayout layout;
	Renderer.Callback callback;
	
	Renderer(ForceDirectedLayout layout, Renderer.Callback callback) {
		this.layout = layout;
		this.callback = callback;
	}
}

File: springy-swift/springy/Renderer.swift
//
//  Renderer.swift
//  springy
//
//  Created by 김장환 on 1/16/15.
//  Copyright (c) 2015 김장환. All rights reserved.
//

import Foundation

class Renderer:GraphListener {
    var layout:ForceDirectedLayout
    let clear: () -> ()
    let drawEdge: (Edge, Point, Point) -> ()
    let drawNode: (Node, Point) -> ()
    let onRenderStop: () -> ()
    let onRenderStart: () -> ()

    init(layout:ForceDirectedLayout,
            clear: () -> () = {},
            drawEdge: (Edge, Point, Point) -> () = { (_,_,_) in return },
            drawNode: (Node, Point) -> () = { (_,_) in return },
            onRenderStart: () -> () = {},
            onRenderStop: () -> () = {}
    ) {
        self.layout = layout
        self.clear = clear
        self.drawEdge = drawEdge
        self.drawNode = drawNode
        self.onRenderStop = onRenderStop
        self.onRenderStart = onRenderStart
    }
    
    func graphChanged() {
//        self.start()
    }

}